Automattic/woocommerce-payments
# Improve PHP code coverage

Fixes #5977 

#### Changes proposed in this Pull Request

- Add unit tests to files that lack code coverage
- Fix search by order id on Risk Review/Blocked tabs on the transactions page

| Before | After |
|--------|--------|
| Code coverage: `60.31%`  | Code coverage: `62.12%` |
| <img width="826" alt="Screenshot 2023-04-04 at 21 40 36" src="https://user-images.githubusercontent.com/16882226/229959252-adadc834-950b-41fe-85bb-075317f1d656.png"> | <img width="824" alt="image" src="https://user-images.githubusercontent.com/16882226/231630037-f9f17a86-4700-4c16-85fa-37efac2d68b8.png"> | 

The 100% coverage files jumped from 68% to 71% and the 0-10% from almost 20% to around 10%.









#### Testing instructions





* Unit tests should be passing
* Checkout to this branch
* Build the client `npm run start`
* Listen to Stripe webhooks `npm run listen`
* Make sure the version on `woocommerce-payments.php` is at least `5.8.0`
* Make sure the fraud settings feature flag is enabled by running `npm run wp option add wcpay_fraud_protection_settings_active 1`
* Go to **Payments > Settings**, scroll to the Fraud protection section, select the advanced protection level, and click on "Configure"
* Scroll to the "Order items threshold" rule, enable the screening, and set the maximum items to 3
* Save the settings
* Go to your test site, add 4 items to the cart, and perform the checkout – do it a second time (so we can filter it later)
* Go to **Payments > Transactions**, click on the "Risk review" tab
* There should be at least 2 items on that list – click on the first one
* Copy the order number
* Go to **Payments > Transactions**, click on the "Risk review" tab
* Paste the order number into the search input
* There should be a suggestion

<img width="680" alt="image" src="https://user-images.githubusercontent.com/16882226/231628686-99d64d4e-0cc8-46ae-a29f-0398d41c2a5f.png">

* Select the suggested option
* Only the selected order item should appear in the transactions list

<img width="1498" alt="image" src="https://user-images.githubusercontent.com/16882226/231628795-053c1919-6e63-4f0d-80b5-e1ddc299f6d7.png">


-------------------


**Post merge**
